src/pool/srv_pool.c

```diff
@@ -2479,8 +2479,15 @@ pool_prop_read(struct rdb_tx *tx, const struct pool_svc *svc, uint64_t bits,
 	if (bits & DAOS_PO_QUERY_PROP_DATA_THRESH) {
 		d_iov_set(&value, &val, sizeof(val));
 		rc = rdb_tx_lookup(tx, &svc->ps_root, &ds_pool_prop_data_thresh, &value);
-		if (rc != 0)
+		if (rc == -DER_NONEXIST && global_ver < 3) { /* needs to be upgraded */
```

Comment:
@kccain, we generally use some named constant (e.g., a macro) instead of "3", is that right?

Reply:
There are a few examples where we have done this, yes. And I thought of this when initially I was only reviewing this patch. But unfortunately it is not applied in the majority of cases and so I didn't mention it for that reason.

An example is near the top of this file, the constant DAOS_POOL_GLOBAL_VERSION_WITH_SVC_OPS_KVS (3)
It could certainly be done in this patch with the ds_pool_prop_data_thresh property.

Reply:
It is a new property that won't be queried upon downgrade. So it should be fine, no?

Reply:
It's hard to say if such an unknown property will confuse the 2.4 code: One has to check and test.

Reply:
Hence, I think we should not write such a property for a pool that is in the "2.4 mode" even if the engine code is post-2.4. Then we know we'd be fine after a downgrade.